File: jiutian-bigdata/src/main/java/flink/scala/opentsdb/client/HttpClient.scala
package flink.scala.opentsdb.client

import java.io.IOException

import flink.scala.opentsdb.client.ExpectResponse.ExpectResponse
import flink.scala.opentsdb.client.builder.MetricBuilder
import flink.scala.opentsdb.client.request.QueryBuilder
import flink.scala.opentsdb.client.response.{Response, SimpleHttpResponse}


/**
  * Created on 2019-06-17
  *
  * @author :hao.li
  */
trait HttpClient extends Client {
  @throws[IOException]
  def pushMetrics(builder: MetricBuilder, exceptResponse: ExpectResponse): Response

  @throws[IOException]
  def pushQueries(builder: QueryBuilder, exceptResponse: ExpectResponse): SimpleHttpResponse

}

File: jiutian-bigdata/src/main/java/flink/java/opentsdb/client/HttpClient.java
package flink.java.opentsdb.client;



import flink.java.opentsdb.client.builder.MetricBuilder;
import flink.java.opentsdb.client.request.QueryBuilder;
import flink.java.opentsdb.client.response.Response;
import flink.java.opentsdb.client.response.SimpleHttpResponse;

import java.io.IOException;

public interface HttpClient extends Client {

	public Response pushMetrics(MetricBuilder builder,
								ExpectResponse exceptResponse) throws IOException;

	public SimpleHttpResponse pushQueries(QueryBuilder builder,
										  ExpectResponse exceptResponse) throws IOException;
}
